src/Access/ContextAccess.cpp

```diff
@@ -246,6 +246,17 @@ void ContextAccess::calculateAccessRights() const
 
 
 UserPtr ContextAccess::getUser() const
+{
+    auto res = tryGetUser();
+
+    if (likely(res))
+        return res;
+
+    throw Exception(ErrorCodes::UNKNOWN_USER, "User has been dropped");
```

Comment:
If `user` is null that means it's either dropped or this is the global context, the error message should distinguish these cases.

Reply:
Or we could move the function `checkGranteeIsAllowed()` to `ContextAccess` and make sure it doesn't throw when it's called for a `ContextAccess` generated by call `ContextAccess::getFullAccess()`.

Reply:
I meant the error message should be different:
`UNKNOWN_USER + "User has been dropped"`
-OR-
`LOGICAL_ERROR + "No user"`

I think that would be a bit better than always telling that user has been dropped even if it's not true